src/components/screens/questionnaire-screen.tsx

```diff
@@ -81,35 +58,12 @@ const QuestionnaireScreen = () => {
             }
           </Grid>
         </Grid>
-        {mockQuestionnaires.map((questionnaire) => (
-          <Grid item xs={12} key={questionnaire.id} sx={{ mt: 2 }}>
-            <Card sx={{ p: 2 }}>
-              <CardContent>
-                <Typography variant="h6">{questionnaire.title}</Typography>
-                <Typography variant="body2">{questionnaire.description}</Typography>
-                <Box display="flex" alignItems="center">
-                  <Typography variant="body2" sx={{ mr: 1 }}>
-                    {strings.questionnaireScreen.status}
-                  </Typography>
-                  {questionnaire.status}
-                </Box>
-              </CardContent>
-              {adminMode && 
-                <CardActions>
-                  <Button variant="outlined" color="success">
-                    <EditIcon sx={{ color: "success", mr: 2 }} />
-                    {strings.questionnaireScreen.edit}
-                  </Button>
-                  <Button variant="contained" color="secondary">
-                    <DeleteForeverIcon sx={{ color: "red", mr: 2 }} />
-                    {strings.questionnaireScreen.delete}
-                  </Button>
-                </CardActions>
-              }
-            </Card>
-          </Grid>
-        ))}
       </Grid>
+      <Suspense fallback={<CircularProgress/>}>
+        <QuestionnaireTable/>
+      </Suspense>
```

Comment:
Couldn't there just be a renderLoader as in other components? Or do you think suspense is better? Seems to work from in the UI.